examples/apps/spaces/src/fluid-object/spacesItemMap.ts

```diff
@@ -63,28 +90,28 @@ export interface ISpacesItemEntry<T = any> {
 
 const clickerItemEntry: ISpacesItemEntry<ISingleHandleItem> = {
     create: createSingleHandleItem(ClickerInstantiationFactory),
-    getView: getAdaptedViewForSingleHandleItem,
+    getView: getClickerView,
     friendlyName: "Clicker",
     fabricIconName: "Touch",
 };
 
 const codemirrorItemEntry: ISpacesItemEntry<ISingleHandleItem> = {
-    create: createSingleHandleItem(cmfe),
-    getView: getAdaptedViewForSingleHandleItem,
+    create: createSingleHandleItem(codeMirrorFactory),
+    getView: getCodeMirrorView,
     friendlyName: "Code",
     fabricIconName: "Code",
 };
 
 const textboxItemEntry: ISpacesItemEntry<ISingleHandleItem> = {
     create: createSingleHandleItem(CollaborativeText.getFactory()),
-    getView: getAdaptedViewForSingleHandleItem,
+    getView: getCollaborativeTextView,
```

Comment:
Looking at the `spacesStorageView` the idea is to re-create this element for any change on the "shared" object ?

Reply:
Basically whenever the `SpacesStorage` adds or removes a data object from its collection, it raises an event that notifies the `SpacesStorageView` of the change.  `SpacesStorageView` then calls `getViewForItem()` which will ultimately refer back to the `getView()` entries here to retrieve the appropriate view for the data object that was just added.

The main benefit of this approach being that we're able centralize the intimate knowledge of how the data objects are paired with views in the registry, allowing the `SpacesView`, `SpacesStorageView`, `SpacesItemView` to treat the data objects generically and allowing `Spaces` and `SpacesStorage` to be unopinionated about the views used to render their data.

Hopefully there is not too much re-creation of the views (I rely on React to suppress some of that here) but it's possible there's some over-re-creation.  If there is though, it should be relatively benign since the re-created view is separate from the data.